src/NuGet.Core/NuGet.ProjectModel/JsonUtility.cs

```diff
@@ -141,5 +170,36 @@ internal static JToken WriteString(string item)
         {
             return item != null ? new JValue(item) : JValue.CreateNull();
         }
+
+        internal static NuGetVersion ParseNugetVersion(string value)
```

Comment:
https://github.com/NuGet/NuGet.Client/pull/5529#discussion_r1414976725

@davkean @nkolev92  continuing this conversation here. Would we want the cache of version/version range to live only during the parsing of the file?

Reply:
I think defining a great lifetime for these caches would be challenging.

I'm comfortable if this change starts with the assets file read scope, but we should look into expanding it.

Reply:
I love that, so I went ahead and undid the caching stuff and saved a stash of it.